### PR TITLE
Switch head to SUSE Manager 4.0 based on SLE15 SP1

### PR DESCRIPTION
**Pull request by mcalmer**
add the missing pieces to install SUSE Manager 4.0 (head) with SLE 15 SP1.

- add  image definition
- sumaform tools for SLE15 SP1
- Module repository definitions required for SUSE Manager
- change defaults for head
- fix firewall package

### CODE REVIEW

File: modules/libvirt/base/main.tf [hunk 1]
```diff
@@ -23,6 +23,13 @@ resource "libvirt_volume" "sles15_volume" {
   pool = "${var.pool}"
 }
 
+resource "libvirt_volume" "sles15sp1_volume" {
```
Comment (moio) [inline modules/libvirt/base/main.tf]: Please propagate these changes to the `modules/libvirt` directory also to `libvirt/openstack`. Image is already built at IBS:

https://build.suse.de/project/monitor/Devel:Galaxy:Terraform:Images:OpenStack

At this point I am fine if OpenStack support is there, even if not tested.

File: salt/suse_manager_proxy/init.sls [hunk 2]
```diff
@@ -13,7 +13,11 @@ proxy-packages:
       {% else %}
       - patterns-suma_proxy
       {% endif %}
+{% if grains.get('osmajorrelease', None)|int() == 15 %}
```
Comment (moio) [inline salt/suse_manager_proxy/init.sls]: Please propagate this change also to the `suse_manager_server` module.

Comment (mcalmer) [reply salt/suse_manager_proxy/init.sls]: This already exists for server (https://github.com/moio/sumaform/blob/master/salt/suse_manager_server/firewall.sls#L6)

File: modules/libvirt/base/main.tf (hunk 1 above)
Comment (mcalmer) [reply modules/libvirt/base/main.tf]: done

**Pull request merged**